### PR TITLE
Add Checkstyle framework

### LINKED ISSUE #3533: Checkstyle
Why don't we use Checkstlyle? Reviews for PRs like https://github.com/metamx/java-util/pull/55 include a lot of comments on style. This work is for machines.

### PR DESCRIPTION
**Pull request by leventov**
Fixes #3533.

This PR just adds the Checkstyle framework without enforcing any rules yet; they are supposed to be added one-by-one in different PRs.

Checkstyle's XML configs are placed in a directory named `codestyle` because there is an intention to move `druid_intellij_formatting.xml`, `eclipse_formatting.xml` and `eclipse.importorder` there eventually.

### CODE REVIEW

**Comment by b-slim** [comment]
is this using the druid template to check stuff ?

**Comment by leventov** [comment]
@b-slim it doesn't check anything yet. Eventually we should add checks to enforce druid's rules. Two examples are commented in checkstyle.xml.

**Comment by cheddar** [comment]
Is this PR a WIP?  If it's not a WIP, I politely request that we make it a WIP given that it doesn't actually enforce anything yet.

**Comment by leventov** [comment]
@cheddar added some checks. Don't want to make this PR too big though. It's impossible to make checkstyle to enforce the whole Druid's code style in one PR.

**Comment by cheddar** [comment]
👍 

Totally understand that it's not really possible to do everything.  But also, introducing just boilerplate that doesn't do anything is equivalent to just adding cruft.  Thanks for making this check something and fixing all the things.

File: codestyle/checkstyle-suppressions.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied. See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+<suppressions>
+  <!-- See http://checkstyle.sourceforge.net/config_filters.html#SuppressionFilter for examples -->
+</suppressions>
```
Comment (logarithm) [inline codestyle/checkstyle-suppressions.xml]: Should it be a new line in the end of the file?

File: codestyle/checkstyle.xml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    <!-- See http://checkstyle.sourceforge.net/checks.html for examples -->
+
+    <!--<module name="LineLength">-->
+      <!--<property name="max" value="120"/>-->
+    <!--</module>-->
+    <module name="AvoidStarImport"/>
+    <module name="RedundantImport"/>
+    <module name="NeedBraces"/>
+  </module>
+</module>
```
Comment (logarithm) [inline codestyle/checkstyle.xml]: Should it be a new line in the end of the file?

Comment (leventov) [inline codestyle/checkstyle.xml]: @logarithm thanks. Made Checkstyle to check this, btw :)

**Comment by b-slim** [comment]
👍

**Pull request merged**